packages/flutter/lib/src/material/drop_down.dart

```diff
@@ -218,13 +225,21 @@ class _DropDownMenuRouteLayout extends SingleChildLayoutDelegate {
       }
       return true;
     });
+
+    if (route.initialLayout) {
+      route.initialLayout = false;
+      final double scrollOffset = selectedItemOffset - (buttonTop - top);
+      scrollableKey.currentState.scrollTo(scrollOffset);
+    }
+
     return new Offset(buttonRect.left, top);
   }
 
   @override
-  bool shouldRelayout(_DropDownMenuRouteLayout oldDelegate) {
+  bool shouldRelayout(_DropDownMenuRouteLayout<T> oldDelegate) {
     return oldDelegate.buttonRect != buttonRect
-        || oldDelegate.selectedIndex != selectedIndex;
+        || oldDelegate.selectedIndex != selectedIndex
+        || oldDelegate.scrollableKey != scrollableKey;
```

Comment:
These are immutable on the route, right?  Presumably we could just check the route's identity.